### PR TITLE
neomutt/signature: unset if disabled

### LINKED ISSUE #2876: bug: neomutt doesn't reset the signature when changing accounts
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Issue description

I've multiple mail accounts configured, and one of them has no signature. If I change to this account, the signature of the previous account is set. Please add `unset signature` in the account hook, if no signature is specified for the account.

### Maintainer CC

_No response_

### System information

```markdown
error (ignored): error: end of string reached
error: getting status of '/var/src/config': No such file or directory
(use '--show-trace' to show detailed location information)
 - system: `0`
 - host os: `Linux 5.10.109, NixOS, 21.11 (Porcupine)`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.5.0pre20211206_d1aaa7e`
 - channels(toon): `""`
 - channels(root): `""`
 - nixpkgs: `/nix/store/9dmig1pv9njj5kswvs8yvw3qp6b81zkd-source`
```

### PR DESCRIPTION
**Pull request by sumnerevans**
### Description

Closes #2876

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: modules/programs/neomutt.nix [hunk 1]
```diff
@@ -223,9 +223,11 @@ let
 
       # Extra configuration
       ${account.neomutt.extraConfig}
-    '' + optionalString (account.signature.showSignature != "none") ''
```
Comment (rycee) [inline modules/programs/neomutt.nix:226]: An alternative may be to do something like
``` nix
setOption "signature" (
  if account.signature.showSignature == "none" then
    null
  else
    "${pkgs.writeText "signature.txt" account.signature.text}"
)
```
although I don't know if it is particularly better.

File: modules/programs/neomutt.nix [hunk 2]
```diff
@@ -223,9 +223,11 @@ let
...
+    '' + (if (account.signature.showSignature == "none") then ''
```
Comment (rycee) [inline modules/programs/neomutt.nix:226]: If you prefer using this code then can remove the unnecessary parentheses:
```suggestion
    '' + (if account.signature.showSignature == "none" then ''
```

**Comment by rycee** [comment]
Thanks! Looks good to me, just had a minor comment.

**Review by berbiche — APPROVED** [review]

File: modules/programs/neomutt.nix (hunk 1 above)
Comment (sumnerevans) [reply modules/programs/neomutt.nix:226]: I think I'm going to keep it as-is. I don't think that is any better than what I have.

**Comment by teto** [comment]
I dont use neomutt but the problem may arise for other settings too, is there a way to reset settings between accounts, i.e., to start from a clean slate ?

**Pull request merged**